processing/src/main/java/org/apache/carbondata/processing/csvreaderstep/UnivocityCsvParser.java

```diff
@@ -102,8 +102,8 @@ public void initialize() throws IOException {
     parserSettings.setMaxColumns(
         getMaxColumnsForParsing(csvParserVo.getNumberOfColumns(), csvParserVo.getMaxColumns()));
     parserSettings.setNullValue("");
-    parserSettings.setIgnoreLeadingWhitespaces(false);
-    parserSettings.setIgnoreTrailingWhitespaces(false);
+    parserSettings.setIgnoreLeadingWhitespaces(csvParserVo.getTrim());
```

Comment:
hi sujith
i am thinking if user already trim the data with the option setting,then when user query with some space filter,it would no getting result.